File: Tellma/Controllers/DTO/GetAggregateResponse.cs
using System;
using System.Collections;
using System.Collections.Generic;
using Tellma.Data.Queries;

namespace Tellma.Controllers.Dto
{
    public class GetAggregateResponse
    {
        /// <summary>
        /// Contains the result of the aggregate query
        /// </summary>
        public IEnumerable<DynamicRow> Result { get; set; }

        /// <summary>
        /// Contains the ancestors of tree dimensions if any
        /// </summary>
        public IEnumerable<TreeDimensionResult> DimensionAncestors { get; set; }

        public DateTimeOffset ServerTime { get; set; }

        public bool IsPartial { get; set; }
    }


}

File: Tellma.Api.Web/ClientApp/src/app/data/dto/get-aggregate-response.ts
// tslint:disable:variable-name
export type DynamicRow = any[];

export interface DimensionAncestors {
    IdIndex: number;
    MinIndex: number;
    Result: DynamicRow[];
}

export interface GetAggregateResponse {
    Result: DynamicRow[];
    DimensionAncestors: DimensionAncestors[];
    ServerTime: string;
}
